### PR TITLE
Document nullability of newInstanceFromRadioList() methods and its callers

### PR DESCRIPTION
**Pull request by oleg-nenashev**
This is a follow-up to my review of #3969. I just documented the current APIs while I was there

### Proposed changelog entries

* Document nullability of newInstanceFromRadioList() methods and its callers
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

File: core/src/main/java/hudson/DescriptorExtensionList.java [hunk 1]
```diff
@@ -46,7 +46,9 @@
 import java.util.logging.Logger;
 import java.util.concurrent.CopyOnWriteArrayList;
 import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
 
+import net.sf.json.JSONException;
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/DescriptorExtensionList.java]:
```suggestion
```

File: core/src/main/java/hudson/util/DescriptorList.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
 import hudson.model.Descriptor;
 import hudson.model.Descriptor.FormException;
 import jenkins.model.Jenkins;
+import net.sf.json.JSONException;
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/util/DescriptorList.java]:
```suggestion
```

**Review by MarkEWaite — APPROVED** [review]

**Review by res0nance — APPROVED** [review]
Love documentation improvements ❤️ . CI seems to fail due to javadocs though

File: core/src/main/java/hudson/util/DescriptorList.java [hunk 3]
```diff
@@ -167,7 +168,8 @@ public T newInstanceFromRadioList(JSONObject config) throws FormException {
      * by a radio button group. 
      * @param parent JSON, which contains the configuration entry for the radio list
      * @param name Name of the configuration entry for the radio list
-     * @return new instance or {@code null} if none was selected in the radio list
+     * @return New instance.
+     *         {@code null} if none was selected in the radio list or if the value is filtered by a {@link hudson.model.DescriptorVisibilityFilter
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/util/DescriptorList.java]:
```suggestion
     *         {@code null} if none was selected in the radio list or if the value is filtered by a {@link hudson.model.DescriptorVisibilityFilter}
```

File: core/src/main/java/hudson/util/DescriptorList.java [hunk 4]
```diff
@@ -150,7 +150,8 @@ public boolean remove(Object o) {
      * from the structured form submission data posted
      * by a radio button group. 
      * @param config Submitted configuration for Radio List
-     * @return new instance or {@code null} if none was selected in the radio list
+     * @return New instance.
+     *         {@code null} if none was selected in the radio list or if the value is filtered by a {@link hudson.model.DescriptorVisibilityFilter
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/util/DescriptorList.java]:
```suggestion
     *         {@code null} if none was selected in the radio list or if the value is filtered by a {@link hudson.model.DescriptorVisibilityFilter}
```

File: core/src/main/java/hudson/DescriptorExtensionList.java [hunk 5]
```diff
@@ -132,18 +133,31 @@ public D find(Class<? extends T> type) {
     }
 
     /**
-     * Creates a new instance of a {@link Describable}
-     * from the structured form submission data posted
-     * by a radio button group.
+     * Creates a new instance of a {@link Describable} from the structured form submission data posted by a radio button group.
+     * @param config Submitted configuration for Radio List
+     * @return New instance.
+     *         {@code null} if none was selected in the radio list or if the value is filtered by a {@link hudson.model.DescriptorVisibilityFilter
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/DescriptorExtensionList.java]:
```suggestion
     *         {@code null} if none was selected in the radio list or if the value is filtered by a {@link hudson.model.DescriptorVisibilityFilter}
```

**Review by oleg-nenashev — COMMENTED** [review]
Worked in my IDE :(

**Comment by oleg-nenashev** [comment]
I plan to merge it tomorrow if no negative feedback

**Pull request merged**